### PR TITLE
added 'test_added' and 'test_deleted' fields

### PR DESCRIPTION
**Pull request by LakshyAAAgrawal**
Added check based on "ever_backedout" to decide if it is a test commit or real.

Fixes https://github.com/mozilla/bugbug/issues/111

### CODE REVIEW

**Comment by LakshyAAAgrawal** [comment]
Is this approach appropriate?

**Comment by marco-c** [comment]
Mmmh, test changes are not changes made in backed out commits. Where did you get this idea? :smile:

**Comment by marco-c** [comment]
Answering to https://github.com/mozilla/bugbug/issues/111#issuecomment-473938303:
> Hey,
> Really sorry to have got it wrong.
> Should I instead check for the filename in the variable path if it contains 'test' or not using regex, and if it does, then consider them to be test code changes?

You can check the test files here: https://searchfox.org/mozilla-central/search?q=&case=false&regexp=false&path=test
Try to make a rule out of this.
The best option might be to check if one of the path segments is 'test' or 'tests', but we have to check if there are many false positives.

**Comment by LakshyAAAgrawal** [comment]
Hey,
I have added the test for 'test' or 'tests' segment in the 'path' variable. Could you tell me how to decide whether a file within these directories is a test or not, so that I could attempt to make a rule based on them?
Aren't all the files within these directories relevant to testing, and thus be counted as an addition to test?

**Comment by marco-c** [comment]
Let's use this: https://github.com/mozsearch/mozsearch/blob/2e24a308bf66b4c149683bfeb4ceeea3b250009a/router/router.py#L127.

File: bugbug/repository.py [hunk 1]
```diff
@@ -41,6 +41,15 @@ def _init(repo_dir):
 
 
 def _transform(commit):
+    def is_test(p):
```
Comment (marco-c) [inline bugbug/repository.py]: Move the function outside of _transform, rename `p` to `path`.
Add a link to the source code where you got the function.

File: bugbug/repository.py [hunk 2]
```diff
@@ -41,6 +41,15 @@ def _init(repo_dir):
...
+        if '/unit/' in p:
+            return True
+        if 'test' not in p:
+            return False
```
Comment (marco-c) [inline bugbug/repository.py]: The first two ifs are probably unnecessary, I can't see when they would help

Comment (marco-c) [reply bugbug/repository.py]: Maybe they are optimizations, but I think we can remove them in our case.

**Comment by LakshyAAAgrawal** [comment]
Hey,
I have added the is_test function from the [source](https://github.com/mozsearch/mozsearch/blob/2e24a308bf66b4c149683bfeb4ceeea3b250009a/router/router.py#L127) mentioned above.

However, I think I found an issue while making the said changes.
flake8 is testing against 'W504 line break after binary operator' as well as 'W503 line break before binary operator'.
According to the [PEP8 documentation](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator), line break before binary operator should be allowed.

Therefore, I think W503 should be added in ignore, in the file [setup.cfg](https://github.com/mozilla/bugbug/blob/master/setup.cfg).
Should I do it?

**Comment by marco-c** [comment]
I prefer keeping a vanilla configuration (with the line limit being the only exception). Could you fix the problem in the code instead?

**Comment by LakshyAAAgrawal** [comment]
The current code is consistent with the existing flake8 check. But, I think it might lead to issues in the future with other operators like '+' in string concatenation.

File: bugbug/repository.py (hunk 1 above)
Comment (marco-c) [reply bugbug/repository.py]: The second part of the comment still applies

**Comment by marco-c** [comment]
> The current code is consistent with the existing flake8 check. But, I think it might lead to issues in the future with other operators like '+' in string concatenation.

What issues could it lead to?
The only issue I can think of is failing the linting check, but we can always update the formatting of the code to pass the linting check.

**Comment by LakshyAAAgrawal** [comment]
Yes, I agree.
My only thought was that if there were multiple string concatenations like

```python
return('This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text'+'This_is_a_very_long_string_of_text')
```
It could be better typed as
```python
return('This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text'
       +'This_is_a_very_long_string_of_text')
```
However, I see the point with maintaining a vanilla configuration.

**Comment by marco-c** [comment]
There is just one thing left to do, https://github.com/mozilla/bugbug/pull/218#discussion_r267319861.

File: bugbug/repository.py (hunk 1 above)
Comment (LakshyAAAgrawal) [reply bugbug/repository.py]: I have added the link in the commit message:

>  Added function is_test to check whether a path is related to test or not. The conditions for is_test were obtained from https://github.com/mozsearch/mozsearch/blob/2e24a308bf66b4c149683bfeb4ceeea3b250009a/router/router.py#L127

Should I also add it as a comment in the code?

Comment (marco-c) [reply bugbug/repository.py]: Yes, just add a comment saying
# This code was adapted from XXX

**Pull request merged**